src/Filter/ChoiceFilter.php

```diff
@@ -45,15 +46,17 @@ public function filter(ProxyQueryInterface $proxyQuery, $alias, $field, $data)
             return;
         }
 
-        $andX = $this->getWhere($proxyQuery)->andX();
+        if (EqualOperatorType::TYPE_NOT_EQUAL === $type) {
+            $where = $this->getWhere($proxyQuery)->andX();
+        } else {
+            $where = $this->getWhere($proxyQuery)->orX();
+        }
 
         foreach ($values as $value) {
-            if (ChoiceType::TYPE_NOT_CONTAINS === $type) {
```

Comment:
The `if` was using the 3 ChoiceType constants, but the `operator_type` was the EqualType, which provide just 2 constants.